### DISCUSSION TITLE
Tasmota Script question

### DISCUSSION THREAD

**Discussion by Airmr** [arendst/Tasmota#19978] [Q&A]
Could anybody help me in showing me a script to increase a counter with a switch configured in Module configuration.
I would like to add 1 to the counter every time the switch is triggered.
![image](https://github.com/arendst/Tasmota/assets/57837042/c70c5662-b4d4-4e3e-a54b-5a25d167335e)
I am thankfull for any ideas here. Seems simple, but i am stuck after 2 days of reading...

**Comment by pkkrusty** [comment]
Not sure, but wouldn't `rule1 on switch1#state do counter1 +1 endon` work?

**Reply by Airmr** [reply]
Hi, thnx for the tip. I am using Scripting in Tasmota and as far as i know you cant use rules when scripting is enabled. If what i think is wrong, just enter the "rule1 ...." in the console?

**Reply by sfromis** [reply]
It is correct that rules are not available when you use a scripting binary.

**Reply by Airmr** [reply]
So there is now way to use a switch to count up an internal pio counter on the tasmota device? This can just be done with a variable defined and counted up, right?

**Reply by barbudor** [reply]
Very little people are using script
But Tasmota already has a GPIO type Counter that can count by itself. Isn't that enough ?

Otherwise, may be @gemu2015 can be of help when it comes to scripting

**Reply by pkkrusty** [reply]
With scripting, you have lots of options. Why use counter? Because it displays on main GUI? See following example from scripting page:
```
>E
slider=Switch1

if upd[slider]>0
then
print slider updated %slider%
endif

if upd[power]>0
then
print power updated %power%
endif
```
Anything that gives a RESULT json will trigger in >E section so you can capture switch state change. Then just count up any old variable you choose, or if you want to use counter, then do `=> counter1 +1`

Edit* I take that back. Switch state doesn't give RESULT json. So you'll have to poll for it. Example is in scripting docs.

**Reply by Airmr** [reply]
Hi, this => counter1 +1 helped get the counter more accurate. When just using the counter with a reed contact, every "contact" with a magnet counted 2 up. I could get that to 1 with counterdebounce or just divide by 2. But when the magnet stopped at the contact and stayed their while not consuming any gas till the magnet moved again it counted when it stopped and when it started again. So there was an unpredictable number of counts to much.
With the Switch it counts 2 times but when the magnet stops at the reed contact it does not count again when starting to turn again. So dividing the Switch/Counter by 2 gives a more accurate result in counts. 

Thnx a lot for all answers and contributions. It finally solved the issue.
